### PR TITLE
Add race and some construction helpers to Async

### PR DESCRIPTION
**Pull request by evilsoft**
## The race is on
image

This PR addresses [this issue](https://github.com/evilsoft/crocks/issues/281) and adds the following:
* `Async.rejectAfter`
* `Async.resolveAfter`
* `Async.prototype.race`
* `race` pointfree helper

### CODE REVIEW

File: src/Async/index.js [hunk 1]
```diff
@@ -116,6 +150,26 @@ function Async(fn) {
     })
   }
 
+  function race(m) {
```
Comment (karthikiyengar) [inline src/Async/index.js]: Took me a while to wrap my head around this, great stuff 👍😄

File: docs/src/pages/docs/crocks/Async.md [hunk 2]
```diff
@@ -105,11 +105,11 @@ getById(5)
 const cancel = getById(1).fork(
   log('rej'),
   log('res'),
-  () => console.log('cancelled')
+  () => console.log('canceled')
```
[the remark this answers is not in the document]
Comment (evilsoft) [reply docs/src/pages/docs/crocks/Async.md]: I am sorry. what do you mean by that, and what do I need to do to rectify it?

[the remark this answers is not in the document]
Comment (karthikiyengar) [reply docs/src/pages/docs/crocks/Async.md]: It looks like `fork` accepts 3 functions now, doesn't it?

**Comment by evilsoft** [comment]
@karthikiyengar kk. updated, please give it some 👀 when you have a chance.

**Comment by evilsoft** [comment]
image

Got the updates in there for the documentation as requested by @karthikiyengar.
Gonna merge this to get it in there for the next release.

**Comment by karthikiyengar** [comment]
My apologies, got a bit caught up and I couldn't take a look at it.

**Comment by evilsoft** [comment]
@karthikiyengar NBD at all!

**Pull request merged**